specs/networking/libp2p-standardization.md

```diff
@@ -0,0 +1,123 @@
+	mesh_lo: 4
+	// The maximum number of peers in the mesh network before removing some (D_high in the libp2p sepcs).
+	mesh_high: 12
+	// The number of peers to gossip to during a heartbeat (D_lazy in the libp2p sepcs).
+	gossip_lazy: 6 // defaults to `mesh_size`
+	// Time to live for fanout peers (seconds).
+	fanout_ttl: 60
+	// The number of heartbeats to gossip about.
+	gossip_history: 3
+	// Time between each heartbeat (seconds).
+	heartbeat_interval: 1
+)
+```
+
+### Topics
+
+*The Go and Js implementations use string topics - This is likely to be
+updated to topic hashes in later versions - https://github.com/libp2p/rust-libp2p/issues/473*
+
+For Eth2.0 clients, topics will be sent as `SHA2-256` hashes of the topic string.
+
+There are two main topics used to propagate attestations and beacon blocks to
+all nodes on the network.
+
+- The `beacon_block` topic - This topic is used solely for propagating new
+	beacon blocks to all nodes on the networks.
+- The `beacon_attestation` topic - This topic is used to for propagate
+	aggregated attestations to subscribing nodes (typically block proposers) to
+	be included into future blocks. Attestations will be aggregated in their
+	respective subnets before publishing on this topic.
+
+Shards will be grouped into their own subnets (defined by a shard topic). The
+number of shard subnets will be defined via `SHARD_SUBNET_COUNT` and the shard
+`shard_number % SHARD_SUBNET_COUNT` will be assigned to the topic:
+`shard{shard_number % SHARD_SUBNET_COUNT}`.
+
+### Messages
+
+Messages sent across gossipsub are fixed-size length-prefixed byte arrays.
+Each message has a maximum size of 512KB (estimated from expected largest uncompressed
+block size).
+
+The byte array is prefixed with a unsigned 64 bit length number encoded as an
+`unsigned varint` (https://github.com/multiformats/unsigned-varint). Gossipsub messages therefore take the form:
+```
++--------------------------+
+|     message length       |
++--------------------------+
+|                          |
+|       body (<1M)         |
+|                          |
++--------------------------+
+```
+
+The body of the message is an SSZ-encoded object. For the `beacon_block` topic,
+this will be a `beacon_block`. For the `beacon_attestation` topic, this will be
+an `attestation`.
+
+## Eth-2 RPC
+
+#### Protocol Id: `/eth/serenity/beacon/rpc/1`
+
+The [RPC Interface](./rpc-interface.md) is specified in this repository.
+
+## Identify
+
+#### Protocol Id: `/ipfs/id/1.0.0` (to be updated to `/p2p/id/1.0.0`)
+
+The Identify protocol (defined in go - [identify-go](https://github.com/ipfs/go-ipfs/blob/master/core/commands/id.go) and rust [rust-identify](https://github.com/libp2p/rust-libp2p/blob/master/protocols/identify/src/lib.rs))
+allows a node A to query another node B which information B knows about A. This also includes the addresses B is listening on.
```

Comment:
What's the relationship to `discv5`? Is this supposed to be a temporary substitute, a permanent complement, or a replacement?

Reply:
Yep, temporary placeholder. This is currently what lighthouse is using in place of a proper discovery protocol.

Reply:
Let's add an explicit note that this is a placeholder

Reply:
I'm working on discv5 at the moment. Currently, I think this protocol may still be useful for NAT traversal and can be fed into the discv5 protocol. Potentially we don't need it at all. I'll add a placeholder note for now.